File: Mobile/Firebase & Android/add_mob/java/app/src/main/java/andrei/csubb/com/dentistapp/model/Appointment.java
package andrei.csubb.com.dentistapp.model;

import java.util.Objects;

public class Appointment {
    private String id;
    private String date;
    private Integer duration;
    private String patientName;
    private String reason;

    public Appointment() {
    }

    public Appointment(String date, Integer duration, String patientName, String reason) {
        this.date = date;
        this.duration = duration;
        this.patientName = patientName;
        this.reason = reason;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public Integer getDuration() {
        return duration;
    }

    public void setDuration(Integer duration) {
        this.duration = duration;
    }

    public String getPatientName() {
        return patientName;
    }

    public void setPatientName(String patientName) {
        this.patientName = patientName;
    }

    public String getReason() {
        return reason;
    }

    public void setReason(String reason) {
        this.reason = reason;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Appointment that = (Appointment) o;
        return Objects.equals(id, that.id);
    }

    @Override
    public int hashCode() {

        return Objects.hash(id);
    }
}

File: Mobile/Android/kotlin/lab2/app/src/main/java/andrei/csubb/com/dentistryapp/model/Appointment.kt
package andrei.csubb.com.dentistryapp.model

import andrei.csubb.com.dentistryapp.Utils.convertDateToStringFormat
import org.json.JSONObject
import java.util.*

data class Appointment(val id :Long? = null, var startDate: Date,var endDate:Date,var reason:String, var patientName:String){

    fun toJson():String{
        val jsonObject = JSONObject()
        jsonObject.apply {
            id.let {
                this.put("id", id)
            }
            put("startDate", convertDateToStringFormat(startDate))
            put("endDate", convertDateToStringFormat(endDate))
            put("reason", reason)
            put("patientName", patientName)
        }
        return jsonObject.toString()
    }
}